### PR TITLE
Change topic name logic

### LINKED ISSUE #908: [Nox.Lib] Integration Events Set Topic Name
- depends on #882 
- relates to #799 
# Topic Name 
We need to define the topic name  (MassTransit SetEntityNameFormatter) for each of the IE that we can send.
Todo that extend the existing CustomEntityNameFormatter , and set the entityname to be the IE is {Nox.Solution.PlatformId}.{Nox.Solution.Name}.{Trait}

```
internal class CustomEntityNameFormatter : IEntityNameFormatter
{
    public string FormatEntityName<T>() 
    {
        IntegrationEventAttribute? integrationEventAttribute =
            Attribute.GetCustomAttribute(typeof(T), typeof(IntegrationEventAttribute)) as IntegrationEventAttribute;

        if(integrationEventAttribute == null)
        {
            throw new Exception($"The type {c} must have a {nameof(IntegrationEventAttribute)}");
        }
        return ...;
    }
}
```

### PR DESCRIPTION
**Pull request by ILoveMyTortoise**
closes #908

### CODE REVIEW

File: src/Nox.Lib/Infrastructure/Messaging/AzureServiceBus/CustomEntityNameFormatter.cs [hunk 1]
```diff
@@ -1,12 +1,37 @@
...
+        }
+
         public string FormatEntityName<T>()
         {
-            return "test-integration-event";
+            var integrationEventAttribute = typeof(T).GenericTypeArguments[0].GetCustomAttribute<IntegrationEventTypeAttribute>();
+
+            if (typeof(T).GetGenericTypeDefinition() != typeof(NoxMessageRecord<>))
+            {
+                throw new UnknownMessageTypeException($"Unknown message type received. Message type: '{typeof(T)}'. Can't process message.");
```
Comment (rochar) [inline src/Nox.Lib/Infrastructure/Messaging/AzureServiceBus/CustomEntityNameFormatter.cs]:
```suggestion
                throw new UnknownMessageTypeException($"Unknown message type  {typeof(T)} received..");
```

File: src/Nox.Lib/Infrastructure/Messaging/EventNameIsEmptyException.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+﻿namespace Nox.Exceptions
```
Comment (rochar) [inline src/Nox.Lib/Infrastructure/Messaging/EventNameIsEmptyException.cs]:
```suggestion
﻿namespace Nox.Infrastructure.Messaging
```

Comment (rochar) [reply src/Nox.Lib/Infrastructure/Messaging/EventNameIsEmptyException.cs]: I will update the other exceptions if they do not follow this rule

File: src/Nox.Lib/Infrastructure/Messaging/EventTraitIsEmptyException.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
+﻿namespace Nox.Exceptions
```
Comment (rochar) [inline src/Nox.Lib/Infrastructure/Messaging/EventTraitIsEmptyException.cs]:
```suggestion
﻿namespace Nox.Infrastructure.Messaging
```

File: src/Nox.Lib/Infrastructure/Messaging/EventTraitIsEmptyException.cs [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+{
+
+    /// <summary>
+    ///  Nox configuration is invalid
+    /// </summary>
+    public class EventTraitIsEmptyException : Exception
```
Comment (rochar) [inline src/Nox.Lib/Infrastructure/Messaging/EventTraitIsEmptyException.cs]:
```suggestion
    public class EventTraitIsNotFoundException : Exception
```

just a suggestion

File: src/Nox.Lib/Infrastructure/Messaging/UnknownMessageTypeException.cs [hunk 5]
```diff
@@ -0,0 +1,19 @@
+﻿namespace Nox.Exceptions
```
Comment (rochar) [inline src/Nox.Lib/Infrastructure/Messaging/UnknownMessageTypeException.cs]:
```suggestion
﻿namespace ﻿namespace Nox.Infrastructure.Messaging
```

**Review by rochar — CHANGES_REQUESTED** [review]
looks good just a few suggestions

**Review by alexandervlasenko — APPROVED** [review]

**Review by rochar — APPROVED** [review]

**Pull request merged**